src/Middleware/HttpLogging/src/FileLoggerProcessor.cs

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    internal class FileLoggerProcessor : IAsyncDisposable
+    {
+        private const int _maxQueuedMessages = 1024;
+
+        private string _path;
+        private string _fileName;
+        private int? _maxFileSize;
+        private int? _maxRetainedFiles;
+        private int _fileNumber = 1;
+        private TimeSpan _flushPeriod;
+
+        private IOptionsMonitor<W3CLoggerOptions> _options;
+        private readonly BlockingCollection<string> _messageQueue = new BlockingCollection<string>(_maxQueuedMessages);
+        private readonly ILogger _logger;
+        private readonly List<string> _currentBatch = new List<string>();
+        private Task _outputTask;
+        private CancellationTokenSource _cancellationTokenSource;
+
+        public FileLoggerProcessor(IOptionsMonitor<W3CLoggerOptions> options, IHostEnvironment environment, ILoggerFactory factory)
+        {
+            _options = options;
+            var loggerOptions = _options.CurrentValue;
+            _path = loggerOptions.LogDirectory;
+            if (string.IsNullOrEmpty(_path))
+            {
+                _path = Path.Join(environment.ContentRootPath, "logs", DateTimeOffset.Now.ToUnixTimeSeconds().ToString(CultureInfo.InvariantCulture));
+            }
+            _fileName = loggerOptions.FileName;
+            _maxFileSize = loggerOptions.FileSizeLimit;
+            _maxRetainedFiles = loggerOptions.RetainedFileCountLimit;
+            _flushPeriod = loggerOptions.FlushPeriod;
```

Comment:
We should either update this values when the `IOptionsMonitor` notifies us of a change or just use `IOptions` and say we don't support config changes yet.

Reply:
Done (the former)